app/jobs/organization/classroom_visibility_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Organization
+  class ClassroomVisibilityJob < ApplicationJob
+    queue_as :classroom_visibility
+
+    # rubocop:disable SkipsModelValidations
+    def perform(organization, visibility)
+      public_classroom = (visibility == "public" ? true : false)
+      assignments = organization.assignments + organization.group_assignments
+      assignments.each do |assignment|
+        assignment.update_attribute("public_repo", public_classroom)
+        change_visibility_in_all_assignment_repos(assignment, public_classroom)
+      end
+    end
+
+    private
+
+    def change_visibility_in_all_assignment_repos(assignment, public_repo)
```

Comment:
`public_repo` doesn't really imply this is a boolean, something like `is_public` may be better.

Reply:
@d12 you're right, I used a bad example. Consider old records which haven't been validated with newer validators (for example, empty starter code repos) or a template repo which was changed to a regular repo. When updating the public/private attribute with validation it will fail (starter code repo is not a template error).